### PR TITLE
Result display

### PR DESCRIPTION
**Pull request by devashishtyagi**
Add a bunch of support for calling tools and then create demo out of it.

### CODE REVIEW

File: autogen/agentchat/conversable_agent.py [hunk 1]
```diff
@@ -1469,16 +1477,16 @@ def execute_function(self, func_call, verbose: bool = False) -> Tuple[bool, Dict
         func_name = func_call.get("name", "")
         func = self._function_map.get(func_name, None)
 
+        input_string = self._format_json_str(func_call.get("arguments", "{}"))
```
Comment (pgarbacki) [inline autogen/agentchat/conversable_agent.py:1480]: Are the changes in `autogen/agentchat` something that can be upstreamed?

**Review by pgarbacki — APPROVED** [review]

File: samples/apps/fireworks-studio/.virtual_documents/notebooks/tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (benjibc) [inline samples/apps/fireworks-studio/.virtual_documents/notebooks/tutorial.ipynb:1]: what is this file?!

**Review by benjibc — APPROVED** [review]

File: samples/apps/fireworks-studio/.virtual_documents/notebooks/tutorial.ipynb (hunk 2 above)
Comment (devashishtyagi) [reply samples/apps/fireworks-studio/.virtual_documents/notebooks/tutorial.ipynb:1]: I don't really know why it existed - seemed like somebody checked in accidentally.

File: autogen/agentchat/conversable_agent.py (hunk 1 above)
Comment (devashishtyagi) [reply autogen/agentchat/conversable_agent.py:1480]: It's pretty horrible return code so maybe not this time.

**Pull request merged**